CMakeLists.txt

```diff
@@ -21,13 +25,31 @@ if (NOT MSVC)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2 -std=c99 -pedantic -Wall")
 endif ()
 
-add_library (utf8proc
+add_library (utf8proc SHARED
   utf8proc.c
   utf8proc.h
 )
 
+set(UTF8PROC_FULL_VERSION "${SO_MAJOR}.${SO_MINOR}.${SO_PATCH}")
+
 set_target_properties (utf8proc PROPERTIES
   POSITION_INDEPENDENT_CODE ON
-  VERSION "${SO_MAJOR}.${SO_MINOR}.${SO_PATCH}"
+  VERSION "${UTF8PROC_FULL_VERSION}"
   SOVERSION ${SO_MAJOR}
 )
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/utf8proc.pc.cmakein
+               ${CMAKE_CURRENT_SOURCE_DIR}/utf8proc.pc @ONLY)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}-config.cmake.cmakein
+               ${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}-config.cmake @ONLY)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}-config-version.cmake.cmakein
+               ${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}-config-version.cmake @ONLY)
+
+install(FILES utf8proc.pc DESTINATION "${INSTALL_PKGCONFIG_DIR}")
+install(FILES utf8proc.h DESTINATION "${INSTALL_INC_DIR}")
+install(TARGETS utf8proc
+        ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
+        LIBRARY DESTINATION "${INSTALL_LIB_DIR}")
+  
+install(FILES ${PROJECT_NAME}-config.cmake ${PROJECT_NAME}-config-version.cmake
+        DESTINATION "${INSTALL_LIB_DIR}/${PROJECT_NAME}") 
```

Comment:
I get the impression that `"${CMAKE_INSTALL_PREFIX}/share/cmake/${PROJECT_NAME}"` is preferred here.  
It makes sense to put it in a `cmake` subdirectory so as not to clutter the `lib` or `share` directory, and it makes sense to use `share` rather than `lib` similar to the `.pc` file.

See e.g. [here](https://stackoverflow.com/questions/30201895/what-is-the-location-to-install-cmake-extension-files-in-ubuntu) and [here](https://cmake.org/cmake/help/v3.2/command/find_package.html).